### PR TITLE
added uwsgi ephemeral for migrate, shell, collectstatic that not retried

### PR DESCRIPTION
**Pull request by meomancer**
@cchristelis 
this PR is about added uwsgi ephemeral

### CODE REVIEW

**Comment by timlinux** [comment]
Nice stuff @meomancer ....we need to port this logic over to other projects too....

File: deployment/docker-compose.yml [hunk 1]
```diff
@@ -67,6 +67,30 @@ uwsgi:
   user: root
   restart: on-failure:5
 
+uwsgi-ephemeral:
```
Comment (cchristelis) [inline deployment/docker-compose.yml]: @meomancer, why didn't you subclass?

File: deployment/docker-compose.yml [hunk 2]
```diff
@@ -67,6 +67,30 @@ uwsgi:
   user: root
   restart: on-failure:5
```
Comment (cchristelis) [inline deployment/docker-compose.yml]: This should maybe rather be always?

**Comment by cchristelis** [comment]
@meomancer, I like it, please have a look at my comments.

File: deployment/docker-compose.yml [hunk 3]
```diff
@@ -67,6 +67,30 @@ uwsgi:
...
 
+uwsgi-ephemeral:
+  # Note you cannot scale if you use conteiner_name
+  container_name: mezzanine-uwsgi-ephemeral
```
Comment (timlinux) [inline deployment/docker-compose.yml]: Also maybe it is better to use generic names - so we can easily port logic like this to other projects. So `uwsgi-ephemeral` might be nicer....

**Comment by meomancer** [comment]
@cchristelis @timlinux 
PR is updated

is it better we make max retries so there will be no zombies?

File: deployment/docker-compose.yml [hunk 4]
```diff
@@ -61,11 +60,23 @@ uwsgi:
     - ./media:/home/web/media:rw
     - ./reports:/home/web/reports
     - ./logs:/var/log/
+  user: root
+
+uwsgi-ephemeral:
+  extends: uwsgi-base
+  container_name: mezzanine-uwsgi-ephemeral
```
Comment (cchristelis) [inline deployment/docker-compose.yml]: Let's remove the container_name, because there might be more than one at any given time.

Comment (meomancer) [reply deployment/docker-compose.yml]: @cchristelis If there is no container name, how do i do to migrate/shell to this container?

Comment (cchristelis) [reply deployment/docker-compose.yml]: So we don't expect it to be running, so exec'ing we will do into mezzanine-uwsgi. 
migration: docker-compose -p $(PROJECT_ID) run uwsgi-ephemeral python manage.py migrate 
shell: docker-compose -p $(PROJECT_ID) run uwsgi-ephemeral python manage.py shell

**Comment by meomancer** [comment]
@cchristelis PR updated

**Pull request merged**